extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -56,7 +56,19 @@ void build(RecorderContext recorder,
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, ByteArraySerde.class));
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, FailOnInvalidTimestamp.class));
 
-        nativeLibs.produce(new SubstrateResourceBuildItem(Environment.getJniLibraryFileName("rocksdb")));
+        // for RocksDB, either add linux64 native lib when targeting containers
+        String containerRuntime = System.getProperty("native-image.container-runtime");
+        if (containerRuntime != null) {
+            containerRuntime = containerRuntime.trim();
+
+            if (containerRuntime.equals("docker") || containerRuntime.equals("podman")) {
+                nativeLibs.produce(new SubstrateResourceBuildItem("librocksdbjni-linux64.so"));
+            }
+        }
+        // otherwise the native lib of the platform this build runs on
+        else {
+            nativeLibs.produce(new SubstrateResourceBuildItem(Environment.getJniLibraryFileName("rocksdb")));
```

Comment:
This needs to be testing on windows (yes, I know...).
It's not a blocker as building native executable on windows is still experimental, but it would be nice to know if this works.

Reply:
Uff, I have no Windows machine at my disposal :(

But wasn't native Windows support only added in 19.0? Is Quarkus on this one already?